sky/utils/common_utils.py

```diff
@@ -697,3 +697,26 @@ def truncate_long_string(s: str, max_length: int = 35) -> str:
     if len(prefix) < max_length:
         prefix += s[len(prefix):max_length]
     return prefix + '...'
+
+
+def get_gpu_resource_name(name: str = 'nvidia.com/gpu'):
+    """Get the GPU resource name.
+
+    The function first checks for an environment variable.
+    If defined, it uses its value; otherwise, it returns the default value.
+
+    Args:
+        name (str): Default GPU resource name, default is "nvidia.com/gpu".
+
+    Returns:
+        str: The selected GPU resource name.
+    """
+    # retrieve GPU resource name from environment variable,
+    # can be nvidia.com/gpu-h100 if it's customized by the device plugin
+    custom_name = os.getenv('CUSTOM_GPU_RESOURCE_NAME')
+
+    # If the environment variable is not defined, return the default name
+    if custom_name is None:
+        return name
+
+    return custom_name
```

Comment:
Can we move this to `sky.provision.kubernetes.utils` since only k8s relies on this method?